### PR TITLE
EODHP-540 - add pre-commit

### PR DESCRIPTION
**Pull request by hcollingwood**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+dockerbuild:
+	DOCKER_BUILDKIT=1 docker build -t ${IMAGENAME}:${VERSION} .
+
+dockerpush: dockerbuild testdocker
+	docker tag ${IMAGENAME}:${VERSION} ${DOCKERREPO}/${IMAGENAME}:${VERSION}
+	docker push ${DOCKERREPO}/${IMAGENAME}:${VERSION}
+
+test:
+	./venv/bin/ptw configscanning
```
Comment (ahaywardtvuk) [inline Makefile:14]: For this to work we need `pytest-watcher` in the dev dependencies.

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+	./venv/bin/pytest
+
+ruff:
+	./venv/bin/ruff check .
+
+black:
+	./venv/bin/black .
+
+isort:
+	./venv/bin/isort . --check --diff --profile black
```
Comment (ahaywardtvuk) [inline Makefile]: isort also needs to be in the dev dependencies for this to work.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,59 @@
+.PHONY: dockerbuild dockerpush test testonce ruff black lint isort pre-commit-check requirements-update requirements setup
+VERSION ?= latest
+IMAGENAME = eodhp-web-presence
```
Comment (ahaywardtvuk) [inline Makefile:3]: This image name needs an update.

**Review by ahaywardtvuk — CHANGES_REQUESTED** [review]

File: Makefile (hunk 2 above)
Comment (hcollingwood) [reply Makefile]: Added

File: Makefile (hunk 3 above)
Comment (hcollingwood) [reply Makefile]: Good catch

**Review by ahaywardtvuk — APPROVED** [review]

**Pull request merged**